src/formatDate.js

```diff
@@ -7,8 +7,50 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
```

Comment:
The logic for determining the separator assumes that the last element of the format arrays is the separator. This might not always be the case. Consider explicitly defining the separator or ensuring the format arrays are structured as expected.